rust/worker/src/blockstore/arrow/block/types.rs

```diff
@@ -92,6 +92,126 @@ impl Block {
         None
     }
 
+    pub fn get_prefix<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>>(
+        &'me self,
+        prefix: &str,
+    ) -> Option<Vec<(&str, K, V)>> {
+        let prefix_array = self
+            .data
+            .column(0)
+            .as_any()
+            .downcast_ref::<StringArray>()
+            .unwrap();
+        let mut res: Vec<(&str, K, V)> = vec![];
+        for i in 0..self.data.num_rows() {
+            let curr_prefix = prefix_array.value(i);
+            if curr_prefix == prefix {
+                res.push((
+                    curr_prefix,
+                    K::get(self.data.column(1), i),
+                    V::get(self.data.column(2), i),
+                ));
+            }
+        }
+        return Some(res);
+    }
+
+    pub fn get_gt<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>>(
+        &'me self,
+        prefix: &str,
+        key: K,
+    ) -> Option<Vec<(&str, K, V)>> {
+        let search_key_composite = CompositeKey::new(prefix.to_string(), key.clone());
+        let prefix_array = self
+            .data
+            .column(0)
+            .as_any()
+            .downcast_ref::<StringArray>()
+            .unwrap();
+        let mut res: Vec<(&str, K, V)> = vec![];
+        for i in 0..self.data.num_rows() {
+            let curr_prefix = prefix_array.value(i);
+            let curr_key = K::get(self.data.column(1), i);
+            let curr_key_composite = CompositeKey::new(curr_prefix.to_string(), curr_key.clone());
```

Comment:
can e do this comparison on the fields without relying on composite key? this results in a needless clone and bundle of the subfields